src/matrices.rs

```diff
@@ -232,6 +236,27 @@ impl<T: MatrixNumber> Matrix<T> {
         // We do a little trick and apply `self` twice, as it is more memory efficient
         self.checked_operation_on_two(self, |a, _| operation(a))
     }
+
+    pub fn checked_pow(&self, mut exponent: usize) -> anyhow::Result<Self> {
+        let shape = self.get_shape()?;
+        if shape.0 != shape.1 {
```

Comment:
I would change it to deconstruct `self.get_shape()` into `(h, w) = self.get_shape()`, as we never need the `shape` itself.

Reply:
fine!